tests/Library/Vanilla/AliasProviderTraitTest.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * @author Adam Charron <adam.c@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace VanillaTests\Library\Vanilla;
+
+use VanillaTests\Fixtures\Aliases\ExtendsNewClass;
+use VanillaTests\Fixtures\Aliases\NewClass;
+use VanillaTests\Fixtures\Aliases\TestAliasLoader;
+use VanillaTests\IsolatedTestCase;
+
+/**
+ * Tests for the alias provider trait.
+ */
+class AliasProviderTraitTest extends IsolatedTestCase {
+    /**
+     * Prepare each instance of the test.
+     *
+     * - Disable deprecated notices for the process.
+     * - Register the autoloader for our process.
+     * - Verify that the we are starting with a fresh set of classes. Eg. The classes we're testing have not been
+     * autoloaded yet. $runTestInSeparateProcess should handle this.
+     */
+    public static function setUpBeforeClass() {
+
+        // Our alias autoloader throws deprecated notices. Disable the error reporting for these tests.
+        // Since this test runs in a separate process it shouldn't affect the rest of the tests.
+        error_reporting(E_ALL ^ E_USER_DEPRECATED);
+        spl_autoload_register([TestAliasLoader::class, 'autoload']);
+        self::assertFreshAutoload(NewClass::class);
+        self::assertFreshAutoload(ExtendsNewClass::class);
+    }
+
+    /**
+     * Verify that loading the old class loads the new class through the autoloader.
+     */
+    public function testSimpleAliasAutoload() {
+        new \VanillaTests\OldClass();
```

Comment:
Would it be worth capturing this instance in a variable and asserting it's actually an instance of `NewClass`, instead of only making sure the class was available after this object was instantiated?

Reply:
The fact that is actually an instance of the class is the one line of description the class_alias call has.

http://docs.php.net/manual/da/function.class-alias.php
> Creates an alias named alias based on the defined class original. The aliased class is exactly the same as the original class.

This test function is specifically to test the autoloader. The inheritance checks below kind of cover then case your describing, or the trickier edge case at least.

Reply:
It pains me to see an object instantiated, but not referenced. That's my own personal hangup, though. You are correct in that `assertClassExtendsClass` isn't exactly doing what I'm asking, but it's probably close enough to be practical.